google/earthenterprise
# Python 3 Migration WMS

Rectifies issues with WMS and Python 3.

To test:
1. Build and Install (Build and install according to the build docs (https://github.com/tst-cjeffries/earthenterprise/blob/python_3_wms_bug/earth_enterprise/BUILD_RHEL_CentOS.md)
2. Run the tutorials.
`sudo /opt/google/share/tutorials/fusion/download_tutorial.sh`
`sudo cp earthenterprise/earth_enterprise/src/fusion/tools/gee_test/tutorial_files/providers.xml /gevol/assets/.userdata/`
`sudo env "PATH=$PATH:/opt/google/bin" earthenterprise/earth_enterprise/src/fusion/tools/gee_test/run_tutorial.sh`
Then wait until the tutorials are finished building. This can be monitored in with the status scripts in that same directory.
`sudo env "PATH=$PATH:/opt/google/bin" earthenterprise/earth_enterprise/src/fusion/tools/gee_test/publish_tutorial.sh`
3. Go to the admin page and republish `SF_2d_Merc` with WMS enabled.
4. Try the GetCapabilties Request:
`curl "http://tstserver.vm/SFDb_3d_WMS/wms?version=1.3.0&request=GetCapabilities&service=WMS"`
5. Verify that the returned XML looks correct.
6. Open Earth Client and connect to http://kh.google.com
7. Go to the “Add” menu at the top and click “Image Overlay”
8. Give your overlay a name
9. Click the “Refresh” tab, then “WMS Paramters”
10. Click the “Add…” button at the top, paste in the URL of the WMS database you just published, and click OK.
11. Click “Proceed” if it asks about an untrusted certificate (possibly multiple times)
12. Click “[O] Imagery”, then “Add->”.
13. Click OK twice.
14. The WMS overlay should appear on top of the basemap. Verify that it has the expected content, and remember that it is very slow. If a greyed out area with a giant arrow appears instead, right click on the new layer you added and click “Refresh”. This should force the imagery to appear.
15. Open QGIS
16. Go to Layer > Add Layer > Add WMS ... Layer
17. Create the Layer with the correct url.
18. Add imagery as a JPEG layer
19. Add vector info as a PNG layer.
20. Verify that both layers can be viewed.
